councilmatic_core/management/commands/import_data.py

```diff
@@ -3103,11 +3098,15 @@ def executeTransactionList(self, query_list, args_list):
             for query, args in zip(query_list, args_list):
                 self.connection.execute(query, *args)
 
-    # OCD API has intermittently thrown 502 errors; only proceed when receiving an 'ok' status
-    def _get_response(self, url):
-        response = session.get(url)
+    # OCD API has intermittently thrown 502 and 504 errors; only proceed when receiving an 'ok' status.
+    def _get_response(self, url, params=None, timeout=60, error=True, **kwargs):
+        response = session.get(url, params=params, timeout=timeout, **kwargs)
+        
         if response.ok:
             return response
-        else:
-            self.log_message('WARNING: {url} returned a bad response - {status}'.format(url=url, status=response.status_code), style='ERROR')
+        message = '{url} returned a bad response - {status}'.format(url=url, status=response.status_code)
+        if not error:
+            self.log_message('WARNING: {0}'.format(message), style='ERROR')
             return None
+
+        raise requests.exceptions.HTTPError('ERROR: {0}'.format(message))
```

Comment:
Does `import_data` fail if this exception is raised? Is that what you want?

Reply:
Yes, I'd like the import script to fail. 

However, there are a couple cases where it should not (e.g., when getting the boundary shape - in that case, it should just log a Warning). I added the `error` arg for that reason.